pkgs/applications/editors/helix/default.nix

```diff
@@ -1,33 +1,27 @@
-{ fetchFromGitHub, lib, rustPlatform, installShellFiles, makeWrapper, callPackage }:
+{ fetchzip, lib, rustPlatform, installShellFiles, makeWrapper }:
 
-let
+rustPlatform.buildRustPackage rec {
+  pname = "helix";
   version = "23.05";
 
-  src = fetchFromGitHub {
-    owner = "helix-editor";
-    repo = "helix";
-    rev = "${version}";
-    hash = "sha256-Ws9uWtZLvTwL5HNonFr4YwyPoTU8QlCvhs6IJ92aLDw=";
-  };
-
-  grammars = callPackage ./grammars.nix { };
-in rustPlatform.buildRustPackage {
-  inherit src version;
-
-  pname = "helix";
   # This release tarball includes source code for the tree-sitter grammars,
   # which is not ordinarily part of the repository.
-  cargoSha256 = "sha256-/LCtfyDAA2JuioBD/CDMv6OOxM0B9A3PpuVP/YY5oF0=";
+  src = fetchzip {
+    url = "https://github.com/helix-editor/helix/releases/download/${version}/helix-${version}-source.tar.xz";
+    hash = "sha256-3ZEToXwW569P7IFLqz6Un8rClnWrW5RiYKmRVFt7My8=";
+    stripRoot = false;
+  };
+
+  cargoHash = "sha256-/LCtfyDAA2JuioBD/CDMv6OOxM0B9A3PpuVP/YY5oF0=";
 
   nativeBuildInputs = [ installShellFiles makeWrapper ];
```

Comment:
`git` must be available during the build: we use the `git` binary to check the revisions of the grammar repositories and fetch them if they're not what we expect.

The grammar sources are checked out at the proper revisions in the source tarball so I believe the "No such file or directory" is about the git binary instead.

I haven't looked through the nixpkgs history but maybe `git` was available to buildRustPlatform builds in the past and was removed while nixpkgs handled the grammar fetching itself?

Reply:
> I haven't looked through the nixpkgs history but maybe `git` was available to buildRustPlatform builds in the past and was removed while nixpkgs handled the grammar fetching itself?

Looks like git was available via cargoLock that we were using for tree-sitter. https://github.com/NixOS/nixpkgs/pull/232621/files#diff-ca5625244fa586a1ce711c4fffc9a3d07f382e065458ab3c7d588d2950409cf3L19

Thank you @the-mikedavis.